src/NServiceBus.RabbitMQ/Routing/DefaultRoutingKeyConvention.cs

```diff
@@ -21,6 +22,12 @@ static string GetRoutingKey(Type type, string key = "")
             var interfaces = type.GetInterfaces()
                 .Where(i => !IsSystemType(i) && !IsNServiceBusMarkerInterface(i)).ToList();
 
+            if (interfaces.Count > 1)
+            {
+                Logger.WarnFormat(@"The direct routing topology may not deliver message types with interfaces in their inheritance hierarchy. Refrain from using interface
```

Comment:
Also, "Refrain from using interface for routing purposes.", has a typo. It should be "interfaces". 

However, I'm wondering if this should be changed, to something more specific about not using interfaces for message that are published, because sending doesn't care.